pkg/integrations/windows_exporter/config.go

```diff
@@ -0,0 +1,132 @@
+func (c *Config) ConvertToMap() map[string]string {
+	configMap := make(map[string]string)
+	mapToConfig(c, configMap)
+	return configMap
+}
+
+type ExchangeConfig struct {
+	EnabledList *string `yaml:"enabled_list" exporter:"collectors.exchange.enabled"`
+}
+
+type IISConfig struct {
+	SiteWhiteList *string `yaml:"site_whitelist" exporter:"collector.iis.site-whitelist"`
+	SiteBlackList *string `yaml:"site_blacklist" exporter:"collector.iis.site-blacklist"`
+	AppWhiteList  *string `yaml:"app_whitelist" exporter:"collector.iis.app-whitelist"`
+	AppBlackList  *string `yaml:"app_blacklist" exporter:"collector.iis.app-blacklist"`
+}
+
+type TextFileConfig struct {
+	TextFileDirectory *string `yaml:"text_file_directory" exporter:"collector.textfile.directory"`
+}
+
+type SMTPConfig struct {
+	WhiteList *string `yaml:"whitelist" exporter:"collector.smtp.server-whitelist"`
+	BlackList *string `yaml:"blacklist" exporter:"collector.smtp.server-blacklist"`
+}
+
+type ServiceConfig struct {
+	Where *string `yaml:"where_clause" exporter:"collector.service.services-where"`
+}
+
+type ProcessConfig struct {
+	WhiteList *string `yaml:"whitelist" exporter:"collector.process.whitelist"`
+	BlackList *string `yaml:"blacklist" exporter:"collector.process.blacklist"`
+}
+
+type NetworkConfig struct {
+	WhiteList *string `yaml:"whitelist" exporter:"collector.net.nic-whitelist"`
+	BlackList *string `yaml:"blacklist" exporter:"collector.net.nic-blacklist"`
+}
+
+type MSSQLConfig struct {
+	EnabledClasses *string `yaml:"enabled_classes" exporter:"collectors.mssql.classes-enabled"`
+}
+
+type MSMQConfig struct {
+	Where *string `yaml:"where_clause" exporter:"collector.msmq.msmq-where"`
+}
+
+type LogicalDiskConfig struct {
+	WhiteList *string `yaml:"whitelist" exporter:"collector.logical_disk.volume-whitelist"`
+	BlackList *string `yaml:"blacklist" exporter:"collector.logical_disk.volume-blacklist"`
+}
+
+// This is to create a map from the config using the exporter struct tag as the key. This is to be ONLY used on pointer configs
+// and pointer strings. This is because we need to know if a value was set so that we don't overwrite defaults.
+func mapToConfig(config interface{}, cm map[string]string) {
+	if config == nil || (reflect.ValueOf(config).Kind() == reflect.Ptr && reflect.ValueOf(config).IsNil()) {
+		return
+	}
+	t := reflect.TypeOf(config)
+	v := reflect.ValueOf(config)
+	if t.Kind() != reflect.Ptr {
+		return
+	}
+	fieldCount := t.Elem().NumField()
+	for i := 0; i < fieldCount; i++ {
+		valueField := v.Elem().Field(i)
+		typeField := t.Elem().Field(i)
+		en := typeField.Tag.Get("exporter")
+		// Currently agent only support translating string pointer vars
+		if en != "" && valueField.Kind() == reflect.Ptr && valueField.Elem().Kind() == reflect.String {
+			cm[en] = valueField.Elem().String()
+			continue
+		}
+		if valueField.Kind() == reflect.Ptr {
+			mapToConfig(valueField.Interface(), cm)
+		}
+	}
+}
```

Comment:
In cases like these, I'd prefer to avoid reflection for a simpler, more verbose solution, especially recursive reflection. Can we find a way to retrieve the collector configs and manually map to their settings for each of our config structs? I can kind of imagine something like this:

```go
type LogicalDiskConfig struct {
  WhiteList string `yaml:"whitelist"` 
  BlackList string `yaml:"blacklist"` 
}

func (c LogicalDiskConfig) Sync(v interface{}) {
  if other, ok := v.(*exporter.LogicalDiskConfig); ok {
    other.WhiteList = c.WhiteList 
    other.BlackList = c.BlackList 
  }
}
```

This is more verbose and repetitive, but I believe it's simpler and also means we don't have to have a 1:1 mapping of fields.